rules/type-error.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const tcIdentifiers = new Set([
+	'isArguments',
+	'isArray',
+	'isArrayBuffer',
+	'isArrayLike',
+	'isArrayLikeObject',
+	'isBoolean',
+	'isBuffer',
+	'isDate',
+	'isElement',
+	'isEmptyObject',
+	'isError',
+	'isFinite',
+	'isFrozen',
```

Comment:
I agree with @jfmengels. The reason we should throw a `TypeError` is because there is a mismatch in type. `Object.isFrozen(obj)` does not tell us anything about the type and is not meant to check a type. So imo they should be removed.